torch/optim/adamw.py

```diff
@@ -304,7 +304,7 @@ def adamw(
     See :class:`~torch.optim.AdamW` for details.
     """
 
-    if not all(isinstance(t, torch.Tensor) for t in state_steps):
+    if not torch._utils.is_compiling() and not all(isinstance(t, torch.Tensor) for t in state_steps):
```

Comment:
Wait I'm confused why this one is added--wouldn't state_steps still be full of singleton tensors?

Reply:
Yeah this assert is slow in dynamo. I can check this internally if you prefer, but I honestly felt like this assert isn't needed as much anymore anyway? In the usual case, the optimizer itself sets these tensors up, this is only if the user is directly setting up the optimizer states and calling this method manually.

Reply:
wait so is the only reason this is skipped cuz it’s slow? it doesn’t break any tests?

this check was introduced to ensure steps are tensors and no longer python scalars cc @mikaylagawarecki to confirm—will torch.compile ensure this somewhere down the line? as tracing also requires step be tensors right?

if the sole reason is cuz tracing this is slow, that’s a good thing to point out in a comment

Reply:
@janeyx99 yep that's right the reasoning for adding this check at that point was if I recall correctly 

- these functions were moved from [`torch/optim/_functional.py`](https://github.com/pytorch/pytorch/blob/2a5aaf1c49f18e91c911d6020b71a4201c734eb4/torch/optim/_functional.py) and were technically private APIs that we don't offer a contract on
- these private APIs might be used by internal so we wanted to fail instead of causing silent breakages

Reply:
Yeah the only reason I skipped this is because it's slow. I can add a comment